lib/seattleflu/db/cli/command/longitudinal.py

```diff
@@ -0,0 +1,403 @@
+
+def parse(baseline_filename, weekly_filename, survey_filename, output):
+    """
+    Process and insert longitudinal childcare data from SCH.
+
+    Drops records with no encounter date that result after reshaping from wide
+    to long format.
+
+    All childcare records parsed are output to stdout as newline-delimited JSON
+    records.  You will likely want to redirect stdout to a file.
+    """
+    lnginal_records = format_and_merge_data(baseline_filename, weekly_filename,
+                                            survey_filename)
+
+    lnginal_records = weekly_assessments_wide_to_long(lnginal_records)
+    lnginal_records = select_all_that_apply_wide_to_long(lnginal_records)
+
+    lnginal_records = duplicate_audere_keys(lnginal_records)
+    lnginal_records = insert_static_values(lnginal_records)
+    lnginal_records = create_identifiers(lnginal_records)
+
+    lnginal_records = lnginal_records[lnginal_records.encountered.notnull()]
+
+    barcode_quality_control(lnginal_records, output)
+
+    dump_ndjson(lnginal_records)
+
+
+def format_and_merge_data(baseline_filename: str, weekly_filename: str,
+                          survey_filename: str) -> pd.DataFrame:
+    """
+    For each given filename, loads the related data, performs minimal processing
+    and renaming, and merges each dataset into one DataFrame that is returned.
+    """
+    baseline_records = load_data(baseline_filename)
+    weekly_records = load_data(weekly_filename)
+    survey_records = load_data(survey_filename)
+
+    baseline_records = rename_baseline_columns(baseline_records)
+
+    survey_records = rename_stub(survey_records, regex='^household_(?!ill)',
+                                 current_stub='household',
+                                 desired_stub='household_sx')
+
+    return merge_data(baseline_records, weekly_records, survey_records)
+
+
+def load_data(filename: str) -> pd.DataFrame:
+    """
+    Loads longitudinal childcare data from SCH from a given *filename.
+
+    Removes leading and trailing whitespace from string columns in the
+    underlying data.
+    """
+    df = pd.read_csv(filename)
+    df = trim_whitespace(df)
+    df = add_metadata(df, filename)
+
+    return df
+
+
+def merge_data(baseline_records: pd.DataFrame, weekly_records: pd.DataFrame,
+               survey_records: pd.DataFrame) -> pd.DataFrame:
+    """
+    Given three DataFrames (*baseline_records*, *weekly_records*, and
+    *survey_records*) of longitudinal childcare data, merges them on the unique
+    person identifier (pid).
+
+    Raises a :class:`AssertionError` if the pid columns are missing in any
+    DataFrame or if there are duplicated columns between any two DataFrames
+    other than the pid and '_metadata' columns (added by a previous process).
+
+    The '_metadata' columns in *weekly_records* and *survey_records* receive
+    suffixes of '_followup' and '_survey' after merging, respectively.
```

Comment:
FWIW, the ReStructured Text syntax for a literal bit of code is double backticks, e.g.

```
suffixes of ``_followup`` and ``_survey``
```